src/components/dashboard/health-record/Labs.js

```diff
@@ -21,28 +21,55 @@ export default class Labs extends Component {
   render() {
     if (this.props.labs.length === 0) return <div className="labs no-entries">No entries.</div>;
 
-    // test data - TODO: hook up to actual lab data
-    let data = [
-      { 'value': 200, 'date': '1977-12-16T14:40:51-05:00' },
-      { 'value': 100, 'date': '2017-12-16T14:40:51-05:00' },
-      { 'value': 300, 'date': '2009-12-16T14:40:51-05:00' },
-      { 'value': 500, 'date': '2001-12-16T14:40:51-05:00' }
-    ];
-
-    // test data - TODO: hook up to actual lab data
-    let referenceRanges = [
-      { low: 450, high: 'max', assessment: 'high' },
-      { low: 100, high: 450, assessment: 'average' },
-      { low: 0, high: 100, assessment: 'low' }
-    ];
-
     return (
       <div className="labs">
-        <LineGraph title={"Hemoglobin (%)"} data={data} referenceRanges={referenceRanges} unit={'10^9/L'} />
+        {this.renderLabGraphs()}
         <VerticalTimeline items={this.labs()} icon="flask" />
       </div>
     );
   }
+
+  renderLabGraphs() {
+    let groups = this.groupLabs();
+    let graphs = [];
+    for (var x in groups) {
+      let group = groups[x];
+      graphs.push(<LineGraph  key={x} title={group.title}
+        data={group.values}
+        referenceRanges={group.referenceRanges}
+        unit={group.unit} />);
+    }
+    return graphs.length > 0 ? graphs : '' ;
+  }
+
+  groupLabs() {
+    let grouped = {};
+    this.props.labs.forEach(function(lab) {
+      let code = getProperty(lab, "code.coding.firstObject.code");
```

Comment:
For lines 48-50, since these are never reassigned, react best practice is to use `const`